### PR TITLE
666: Adding new endpoint with corresponding service method. Also a…

### PR DESCRIPTION
**Pull request by vicozizou**
…dding search box to kafka connect screen with consuming logic for added endopoint.

- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)
Backend
Added new endpoint definition to contract project to create `getFilteredConnectors()` at controller interface and its implementation. New (public) method added to service class to preform the actual filtering by connector name, status or type. Integration tests added as well.

Frontend
Added new searchbox to kafka connect page to be used as a search term. React component logic propagates the action to consumen the new backend's endpoint.

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "X" next to an item)

- [ ] No need to
- [ ] Manually(please, describe, when necessary)
- [x] Unit checks
- [x] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings(e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/KafkaConnectController.java [hunk 1]
```diff
@@ -71,6 +72,18 @@
     return Mono.just(ResponseEntity.ok(kafkaConnectService.getAllConnectors(clusterName)));
   }
 
+  @Override
+  public Mono<ResponseEntity<Flux<FullConnectorInfo>>> getFilteredConnectors(
```
Comment (Tereshin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/KafkaConnectController.java:76]: What's the reason to keep `Flux` inside the `Mono`?

Comment (vicozizou) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/KafkaConnectController.java:76]: Honestly @Tereshin I tried to stick to the same way similar methods do it.

File: kafka-ui-contract/src/main/resources/swagger/kafka-connect-api.yaml [hunk 2]
```diff
@@ -53,6 +53,28 @@ paths:
         500:
           description: Internal server error
 
+  /connectors/filtered:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-connect-api.yaml:56]: Could we add this filter to /connectors endpoint and check it for null value? I don't see any reason to have separate endpoint for filtering. We can provide backward compatibility with current endpoint

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-contract/src/main/resources/swagger/kafka-connect-api.yaml (hunk 2 above)
Comment (vicozizou) [reply kafka-ui-contract/src/main/resources/swagger/kafka-connect-api.yaml:56]: Will do.

Comment (vicozizou) [reply kafka-ui-contract/src/main/resources/swagger/kafka-connect-api.yaml]: Done.

**Review by germanosin — APPROVED** [review]

**Pull request merged**